src/prolong_restrict/pr_ops.hpp

```diff
@@ -117,26 +102,36 @@ Real GradMinMod(const Real fc, const Real fm, const Real fp, const Real dxm,
 
 } // namespace util
 
-struct RestrictCellAverage {
-  template <int DIM>
+struct Restrict {
+  static constexpr bool OperationRequired(TopologicalElement fel,
+                                          TopologicalElement cel) {
+    return fel == cel;
+  }
```

Comment:
I think so. It will always be `fel == cel` for the restriction and shared prolongation. And I think it will always be the submanifold thing for the prolongation operator.